rockin/plugins/conveyor_belt/conveyor_belt_thread.cpp

```diff
@@ -30,19 +30,21 @@
  */
 
 /** Constructor. */
-ConveyorBeltThread::ConveyorBeltThread() :
+ConveyorBeltThread::ConveyorBeltThread(std::string mode) :
         Thread("ConveyorBeltThread", Thread::OPMODE_CONTINUOUS), zmq_context_(NULL), zmq_publisher_(NULL), zmq_subscriber_(NULL), cfg_timer_interval_(40), default_network_interface_("eth0")
 {
+    mode_ = mode;
+    last_status_mutex_ = new fawkes::Mutex();
```

Comment:
Why don't you allocate the mutex on the stack (as a member in the class, but not a pointer)?